### PR TITLE
sys: fix uart0

### LINKED ISSUE #3413: shell: inputs for ng_networking example does not react anymore
seems that
```
commit a7f3a3d17fd40f43ab4cd645116224cd1763012a
Merge: 30b2bf6 49ea7a3
Author: Oleg Hahm <oleg@hobbykeller.org>
Date:   Tue Jul 14 23:09:07 2015 +0200

    Merge pull request #3161 from kaspar030/newlib_factor_out_uart_stdio
   
    sys: factor stdio uart functions out of newlib's syscalls
```
broke something for the `samr21-xpro` board.
The `ng_networking` example doesn't react on inputs from this commit on.
Strange is, that the `tests/shell` does.

Can anyone confirm, or is it just me?

### PR DESCRIPTION
**Pull request by kaspar030**
#3161 broke newlib platforms when module_uart0 was used.

This PR bypasses uart0 to uart_stdio when the latter is linked in.

fixes #3413, tested on native, msba2, iot-lab_M3, samr21-xpro.

(uart0 must die!)

### CODE REVIEW

File: sys/posix/posix_io.c [hunk 1]
```diff
@@ -47,6 +47,9 @@ static int _posix_fileop_data(kernel_pid_t pid, int op, char *buffer, int nbytes
 
 int posix_open(int pid, int flags)
 {
+    if (pid==KERNEL_PID_UNDEF) {
```
Comment (gebart) [inline sys/posix/posix_io.c]: add spaces around ==

**Comment by gebart** [comment]
Fixes the problem for me. Tested examples/ng_networking on mulle. Broken on master (doesn't respond to input), works with this PR.

**Comment by gebart** [comment]
ACK, please squash

**Comment by BytesGalore** [comment]
for me too, tested on samr21-xpro

**Comment by BytesGalore** [comment]
also ACK

**Comment by kaspar030** [comment]
- squashed. Merge at will.

**Comment by BytesGalore** [comment]
an GO

**Pull request merged**